### PR TITLE
Eula commands and prompt

### PR DESCRIPTION
**Pull request by vuil**
### What this PR does / why we need it

Add EULA commands, auto prompting for EULA

    tanzu config eula accept
    - accepts the EULA agreement non-interactively

    tanzu config eula show
    - presents the EULA agreement, captures user's agreement

    env var
    TANZU_CLI_EULA_PROMPT_ANSWER=yes
    mainly used for testing, can be used to simulate that user
    has agreed to the EULA.

    Note that the EULA prompt will be presented automatically
    under most conditions (see root.go for exceptions).
    It is unlikely that the user would need to invoke
    'tanzu eula config show' directly

    Note: the text of the EULA prompt is placeholder and intentionally
    formatted to look as such. Will follow up to confirm on the actual
    text to use.

Change depends on:
https://github.com/vmware-tanzu/tanzu-plugin-runtime/pull/69
The commit to update go.mod will not be included in the final merge, but is required for now until 
https://github.com/vmware-tanzu/tanzu-plugin-runtime/pull/69
is merged.

### Which issue(s) this PR fixes

Fixes #N/A

### Describe testing done for PR

Ran updated unit tests.

verify that neither ceip nor eula prompts appear on the following commands

- tanzu version
- tanzu
- tanzu config set
- tanzu config eula show
- tanzu ceip-participation set
- tanzu config eula

### Release note

```release-note
User will be presented with a prompt to accept EULA. Commands `tanzu config eula show` and `tanzu config eula accept` provided to review the EULA and accept it noninteractively, respectively.
```

### CODE REVIEW

**Review by marckhouzam — COMMENTED** [review]
I've played around with this and I think the UX is good.

The PR needs a conflict-resolving rebase, and CI fails.

Nit: the PR description has the subcommands reversed: `tanzu eula config show`

One thing this brings to mind is if the `ceip-participation` command should be moved under `tanzu config`?

File: pkg/command/eula_test.go [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+		Context("When invoking the show command", func() {
+			It("should return successfully", func() {
+				eulaCmd := newEULACmd()
+				eulaCmd.SetArgs([]string{
+					"show"})
+				err = eulaCmd.Execute()
+				Expect(err).To(BeNil())
+			})
+			It("should show even if EULA has been agreed to", func() {
+			})
```
[the remark this answers is not in the document]
Comment (vuil) [reply pkg/command/eula_test.go:52]: Yes, meant to add a note that I am working on it. Actually the testing of iteractive prompts is rather tricky, so I am exploring options.

**Review by marckhouzam — APPROVED** [review]
LGTM
Thanks @vuil !

**Comment by vuil** [comment]
Thanks @marckhouzam. I repushed with some updates to the quickstart doc pertaining to the automatic prompts.

**Review by marckhouzam — APPROVED** [review]
I keep forgetting about the docs!
Thanks for updating them.

LGTM

**Pull request merged**